### DISCUSSION TITLE
Dynamic Routes on Static Site Generation

### DISCUSSION THREAD

**Discussion by QS03** [vercel/next.js#59705] [App Router]
It seems there is no way to export dynamic routes on static site generation, which is available on page router.

![image](https://github.com/vercel/next.js/assets/62456278/dfd477ec-ae62-495d-82ce-591ebdb1108f)

**Comment by SidKH** [comment]
Hey 👋

Check out https://nextjs.org/docs/app/api-reference/functions/generate-static-params

**Reply by QS03** [reply]
@SidKH  I expect dynamic route on-demand at request time, not at build time.

**Comment by wilsonneto-swe** [comment]
Hello @QS03, exactly what SidKH said.

Check the doc here:
https://nextjs.org/docs/app/api-reference/functions/generate-static-params

For sure there are ways to generate dynamic routes as static, it is a basic feature.

Also, if any of the answers helped you, please mark it as solved.

Also, I`m working on an open source project for a community:
https://github.com/craft-code-club/craft-code-club-blog

We will work only with static pages, generating all the websites as a static output if you would like to check it.

**Reply by QS03** [reply]
I believe `generateStaticParams` is used in case dynamic slugs can be decide on build time. In my case, need dynamic slugs on run-time.
There is no dynamic routes in your open source project BTW.

**Comment by QS03** [comment]
I want **dynamic routes on-demand at request time** instead of **dynamic routes at build time**, which means exactly opposite feature of the paragraph described in Next.js docs.

![image](https://github.com/vercel/next.js/assets/62456278/87ed6e3c-4c8c-4c6d-bf0f-89ee682b1259)
